### PR TITLE
fix(html5) - Fix blue screen when user moved to another breakout room

### LINKED ISSUE #16119: After moving a student to a different breakout room, close the obsolete tab
**Describe the bug**
When you move a viewer from breakout roomA to breakout roomB, the viewer will be stuck seeing 

**To Reproduce**
1. Join a meeting as a moderator M
2. Join the meeting as a viewer V
3. M starts a breakout session and puts V in room1
4. V joins room1 (it opens in a new tab)
5. M selects Manage users from the breakouts 3 dot menu and moves V from room1 to room2

**Expected behavior**
V is joining room2, no longer seeing room1.

**Actual behavior**
V is actually stuck seeing a 403 screen that it was removed from the meeting. The user is supposed to notice that **the tab can be closed** and in the main room V is getting invited to room2

Can we close the tab for room1 automatically?

**Screenshots**
Screenshot from 2022-12-12 16-18-52

**BBB version:**
BigBlueButton 2.6.0-beta.2

### LINKED ISSUE #16284: Breakout room's client crashes when you move yourself to another room.
**Describe the bug**
When you have joined a breakout room and move yourself to another room, your breakout room's client crashes.

**To Reproduce**
1. Create breakout rooms.
2. Join a breakout room.
3. Moderator yourself to another room using "Update Breakout Rooms" modal.

**Expected behavior**
Breakout room's client should show the text notifying about being removed from the room.

**Actual behavior**
Your client that's in the breakout room crashes.

**Screenshots**
crash

**BBB version:**
2.6.0-beta.4

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Firefox
 - Version: 108.0.1

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

It fixes blue screen when changing the breakout room assigned to a certain user and it also fixes some minor issues that were being logged.

### Closes Issue(s)

Closes #16284 
Closes #16119

### More

Make sure the errors are happening before testing, to do it, follow the steps:
 1. Create a meeting;
 2. Enter the meeting as moderator;
 3. Open `Manage users` dropdown menu (the gear icon) > `Create breakout rooms`;
 4. Assign yourself to one of the rooms;
 5. In the `BREAKOUT ROOMS` section click `join room` in the exact same room you've chosen before;
 6. Go back to the browser tab where the main meeting is; 
 7. In the `BREAKOUT ROOMS` section click the three dots dropdown menu > `Manage users`;
 8. Change the room you are in and click `Aply`;
 9. See errors in the breakout room's browser tab;

Since the ending of a meeting is a race condition, the final result of the steps I mentioned could end up in a normal MeetingEnd screen, even in the current `v2.6.x-release` brach. Therefore, in order to reproduce this error properly, one should repeat the steps until they are able to see it happening.

The correct flow as for now on:

https://user-images.githubusercontent.com/69865537/213449905-3996c301-5d01-4a59-8a86-d1a56c238cd9.mp4

To test it, play around with the possibilities of ending up a meeting (either main meeting or breakouts), and see if anything breaks down.

**Edit:** After some discussions, we agreed to simply close the previous breakout tab instead of showing a message informing the user that they have been moved!

### CODE REVIEW

File: akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/breakout/ChangeUserBreakoutReqMsgHdlr.scala [hunk 1]
```diff
@@ -58,7 +58,6 @@ trait ChangeUserBreakoutReqMsgHdlr extends RightsManagementTrait {
 
         //Send notification to moved User
         for {
-          roomFrom <- breakoutModel.rooms.get(msg.body.fromBreakoutId)
           roomTo <- breakoutModel.rooms.get(msg.body.toBreakoutId)
```
Comment (gustavotrott) [inline akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/breakout/ChangeUserBreakoutReqMsgHdlr.scala:61]:
```suggestion
	  roomFrom <- breakoutModel.rooms.get(msg.body.fromBreakoutId)
          roomTo <- breakoutModel.rooms.get(msg.body.toBreakoutId)
```

I suggest to keep the `roomFrom` as a validation step.

File: bigbluebutton-html5/imports/startup/client/base.jsx [hunk 2]
```diff
@@ -261,8 +261,16 @@ class Base extends Component {
...
+    
+    if (( meetingHasEnded || ejected || userRemoved ) && meetingIsBreakout) {
+      Base.setExitReason('breakoutEnded').finally(() => {
+        Meteor.disconnect();
+        window.close();
+      });
+      return null;
+    }
+    
+    if (ejected && !meetingIsBreakout) {
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/startup/client/base.jsx:273]:
```suggestion
    if (ejected) {
```

It's unnecessary to validate `meetingIsBreakout` once breakout users will always go into the previous condition.

File: bigbluebutton-html5/imports/ui/services/auth/index.js [hunk 3]
```diff
@@ -217,6 +218,9 @@ class Auth {
         this.loggedIn = true;
         this.uniqueClientSession = `${this.sessionToken}-${Math.random().toString(36).substring(6)}`;
       })
+      .catch(() => {
+        logger.warn("Failed to validate token, user moved to another breakout room");
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/services/auth/index.js:222]: is this log correct?
it seems that this `catch` can handle other cases different of "user moved to another breakout room".

**Review by gustavotrott — CHANGES_REQUESTED** [review]

File: bigbluebutton-html5/imports/ui/services/auth/index.js (hunk 3 above)
Comment (GuiLeme) [reply bigbluebutton-html5/imports/ui/services/auth/index.js:222]: Yeah, as far as I tested this was the only case I could get, but, if you want to, I could make it more generic, what do you think?

File: bigbluebutton-html5/imports/ui/services/auth/index.js [hunk 4]
```diff
@@ -217,6 +218,9 @@ class Auth {
         this.loggedIn = true;
         this.uniqueClientSession = `${this.sessionToken}-${Math.random().toString(36).substring(6)}`;
       })
+      .catch((err) => {
+        logger.warn("Failed to validate token, error: {}", err);
```
Comment (gustavotrott) [inline bigbluebutton-html5/imports/ui/services/auth/index.js:222]:
```suggestion
        logger.error(`Failed to validate token: ${err.description}`);
```

The placeholder `{}` is used in Scala. Not in Javascript.

**Review by gustavotrott — CHANGES_REQUESTED** [review]

**Review by gustavotrott — APPROVED** [review]
LGTM!

**Pull request merged**